x-pack/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -78,8 +81,28 @@ export class VectorLayer extends AbstractLayer {
     });
   }
 
-  getIcon() {
-    return this._style.getIcon();
+  getCustomIconAndTooltipContent() {
+    const sourceDataRequest = this.getSourceDataRequest();
+    const featureCollection = sourceDataRequest ? sourceDataRequest.getData() : null;
+    if (!featureCollection || featureCollection.features.length === 0) {
```

Comment:
I'd consider moving this check to a new `isDataComplete` method on AbstractLayer.

Then, we could use it in the toc_entry to avoid having the legend-details be expanded when there is no data

![image](https://user-images.githubusercontent.com/1833023/57733011-6c7a2300-766c-11e9-9e7a-aea51f0493c4.png)


https://github.com/thomasneirynck/kibana/blob/206dda7f5decebef7fd3cbfb30cd7e3554c58575/x-pack/plugins/maps/public/components/widget_overlay/layer_control/layer_toc/toc_entry/view.js#L239-L241

Reply:
created https://github.com/elastic/kibana/issues/36577